### PR TITLE
Support smoke testing local docker images

### PR DESCRIPTION
**Pull request by sbayer55**
### Description
Allow smoke test to be run against local images
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: release/smoke-tests/run-smoke-tests.sh [hunk 1]
```diff
@@ -35,10 +41,11 @@ function usage() {
     echo ""
     echo "Required arguments:"
     echo -e "-v TAG_NAME\tSpecify the image tag name such as '1.2'"
-    echo -e "-r REPOSITORY\tSpecify the docker repository name (ex: opensearchstaging or opensearchproject). The tag name will be pointed to '-v' value and 'latest'"
     echo ""
     echo "Optional arguments:"
     echo -e "-h\t\tPrint this message."
+    echo -e "-r REPOSITORY\tSpecify the docker repository name (ex: opensearchstaging or opensearchproject). The tag name will be pointed to '-v' value and 'latest'"
```
Comment (sshivanii) [inline release/smoke-tests/run-smoke-tests.sh:47]: Can I know why this was moved from a required to an optional argument?

**Review by chenqi0805 — COMMENTED** [review]
@sbayer55 Can you specify in the README what one needs to do to test locally built image?

File: release/smoke-tests/run-smoke-tests.sh (hunk 1 above)
Comment (dlvenable) [reply release/smoke-tests/run-smoke-tests.sh:47]: I requested this in order to be able to run smoke tests on local images. This way we can pre-test a local build before asking for a staging release.

**Review by chenqi0805 — APPROVED** [review]

File: release/smoke-tests/run-smoke-tests.sh [hunk 2]
```diff
@@ -82,21 +89,34 @@ while getopts "hv:r:o::" arg; do
...
+    # REPOSITORY is defined
+    export DOCKER_IMAGE="${REPOSITORY}/${IMAGE_NAME}:${TAG_NAME}"
+    docker pull "${DOCKER_IMAGE}" > /dev/null 2> /dev/null
+fi
+
+if ! docker inspect --type=image "${DOCKER_IMAGE}" > /dev/null
 then
     echo "--------------------------------------------------------------------------"
-    echo "Unable to pull image \"${REPOSITORY}/data-prepper:${TAG_NAME}\" are you sure it exists?"
+    echo "Unable to pull image \"${DOCKER_IMAGE}\" are you sure it exists?"
```
Comment (cmanning09) [inline release/smoke-tests/run-smoke-tests.sh:115]: This error message will be inaccurate for a local image which does not exist. An alternative could be: `Unable to find image \"${DOCKER_IMAGE}\" are you sure it exists?`

**Review by chenqi0805 — APPROVED** [review]

**Review by cmanning09 — APPROVED** [review]

**Comment by dlvenable** [comment]
Thanks @sbayer55 !

**Pull request merged**